proposals/compatibility-mode.md

```diff
@@ -197,6 +197,18 @@ generate a validation error.
 other APIs only support the first vertex so only `@interpolation(flat, either)` is supported in
 compatibility mode.
 
+## 18. Introduce new `maxVertexStageStorageBuffers` and `maxVertexStageStorageTextures` limits.
+
+If the number of shader variables of type `storage_buffer` in a vertex shader exceeds the `maxVertexStageStorageBuffers` limit, a validation error will occur at pipeline creation time.
+
+If the number of shader variables of type `texture_storage_1d`, `texture_storage_2d`, `texture_storage_2d_array` and `texture_storage_3d` in a vertex shader exceeds the `maxVertexStageStorageTextures` limit, a validation error will occur at pipeline creation time.
+
+In Compatibility mode, these new limits will have a default of zero. In Core mode, they will default to the same values as the existing `maxStorageBuffersPerShaderStage` and `maxStorageTexturesPerShaderStage`, respectively.
+
+In Compatibility mode, the existing `maxStorageBuffersPerShaderStage` and `maxStorageTexturesPerShaderStage` limit will apply to the Compute stage only.
```

Comment:
Alternative: still apply these limits to all stages, so both limits are enforced: the effective limit is `min(maxStorageBuffersPerShaderStage, maxVertexStageStorageBuffers)`.

I somewhat prefer this because (1) then the limit does what it says, and (2) it reduces the amount of logic that changes when compat is active.

Reply:
I was proposing to add limits to the fragment stage as well so that we can increase the default limit for `maxStorageBuffersPerShaderStage` and `maxStorageTexturesPerShaderStage` in compat to 8 (https://github.com/gpuweb/gpuweb/issues/4721#issuecomment-2304348473 & https://github.com/gpuweb/gpuweb/issues/4721#issuecomment-2318325427) but I don't think there is consensus to do that yet.